courses/models.py

```diff
@@ -309,35 +318,58 @@ def __str__(self):
         return self.title
 
 
-class CourseRunEnrollment(TimestampedModel, AuditableModel):
-    """
-    Link between User and CourseRun indicating a user's enrollment
-    """
+class EnrollmentModel(TimestampedModel, AuditableModel):
+    """Abstract base model for enrollments"""
+
+    class Meta:
+        abstract = True
 
     user = models.ForeignKey(User, on_delete=models.PROTECT)
-    run = models.ForeignKey("courses.CourseRun", on_delete=models.PROTECT)
     company = models.ForeignKey(
         "ecommerce.Company", null=True, on_delete=models.PROTECT
     )
-    edx_enrolled = models.BooleanField(
-        default=False,
-        help_text="Indicates whether or not the request succeeded to enroll via the edX API",
+    change_status = models.CharField(
+        choices=ENROLL_CHANGE_STATUS_CHOICES, max_length=20, null=True, blank=True
     )
     active = models.BooleanField(
         default=True,
         help_text="Indicates whether or not this enrollment should be considered active",
     )
 
-    class Meta:
-        unique_together = ("user", "run")
+    active_objects = ActiveEnrollmentManager()
 
     @classmethod
     def get_audit_class(cls):
-        return CourseRunEnrollmentAudit
+        raise NotImplementedError
 
     def to_dict(self):
         return serialize_model_object(self)
 
+    def reactivate_and_save(self):
+        """Sets an enrollment to be active again and saves it"""
+        self.active = True
+        self.change_status = None
+        return self.save_and_log(self.user)
+
+
+class CourseRunEnrollment(EnrollmentModel):
+    """
+    Link between User and CourseRun indicating a user's enrollment
+    """
+
+    run = models.ForeignKey("courses.CourseRun", on_delete=models.PROTECT)
+    edx_enrolled = models.BooleanField(
+        default=False,
+        help_text="Indicates whether or not the request succeeded to enroll via the edX API",
+    )
+
+    class Meta:
+        unique_together = ("user", "run")
+
+    @classmethod
```

Comment:
while you're in here, this could be `staticmethod`

Reply:
The parent class (`AuditableModel`) implementation is defined as a classmethod. Guessing the class reference is needed in some cases